patchwork-events-entity/src/main/java/net/patchworkmc/mixin/event/entity/MixinLivingEntity.java

```diff
@@ -101,6 +115,65 @@ private void hookHandleFallDamageCancel(float distance, float damageMultiplier,
 		}
 	}
 
+	/**
+	 * Carry over the looting level between the two mixins for the drop method.
+	 *
+	 * <p>
+	 * The drop method has roughly this effect:
+	 *
+	 * <pre>{@code
+	 * protected void drop(DamageSource source) {
+	 *     int lootingLevel = ...
+	 *     // hookDropForCapturePre mixin
+	 *     this.dropEquipment(source, lootingLevel, bl);
+	 *     // FRAME CHOP 3
+	 *     this.dropInventory();
+	 *
+	 *     // Added by forge, MUST be called after dropInventory:
+	 *     sendEvent(lootingLevel);
+	 * }
+	 * }</pre>
+	 *
+	 * And thus we can't access the looting level from the end since it's local has
+	 * been discarded. Thus we store it in a field, with an atomic reference to the
+	 * thread currently using it to prevent silent corruption.
+	 * </p>
+	 */
+	@Unique
+	private final AtomicReference<Thread> dropLevelGuard = new AtomicReference<>(null);
+	@Unique
+	private int dropLootingLevel;
+
+	@Inject(method = "drop", at = @At(value = "FIELD", target = "Lnet/minecraft/entity/LivingEntity;playerHitTimer : I"), locals = LocalCapture.CAPTURE_FAILHARD)
+	private void hookDropForCapturePre(DamageSource src, CallbackInfo info, int lootingLevel) {
+		IForgeEntity forgeEntity = (IForgeEntity) this;
+		forgeEntity.captureDrops(new ArrayList<>());
+
+		// Make sure no other threads are currently in this method first
+		if (!dropLevelGuard.compareAndSet(null, Thread.currentThread())) {
+			throw new ConcurrentModificationException("LivingEntity.drop: Single-threaded local hack used by multiple threads");
+		}
```

Comment:
This is an awful hack -- unless you have a good reason to not, you should just use `ThreadLocal`.